### ISSUE TITLE
Problems with the next word predictions

### ISSUE THREAD

**Issue by ishpiki** [salesforce/awd-lstm-lm#106]
Hi, 
I have one issue related to the next word prediction, because by given word and previous hidden states we could try to predict the next most probable word according to softmax probability distribution. The same problem was discussed here #96.

My code snippet looks like the following:

```
def print_prediction(sents):
    pred_ids = []
    idx2token = corpus.dictionary.idx2word
    predicted_words = []
    model.eval()
    hidden = model.init_hidden(1)
    output, hidden = model(sents, hidden)
    output_new = output.view(output.size(0)*output.size(1), output.size(2))
    softmaxed_head_res = torch.nn.functional.softmax(output_new, dim=-1)
    for i in softmaxed_head_res:
        values, ids = i.max(0)
        word = corpus.dictionary.idx2word[ids]
        predicted_words.append(word)
    prediction = ' '.join(predicted_words)
    print('\toriginal: ', ' '.join([idx2token[i[0]] for i in sents.tolist()]))
    print('\tpredicted: ', prediction)
```

When I did it with the trained model (default settings, wiki-2 dataset)  the result was not so good:

> **original**: an English film , television and theatre actor . He had a guest @-@ starring role on the television series The Bill in 2000 . This was followed by a starring role in the play Herons written by Simon Stephens , which was performed in 2001 at the Royal Court Theatre . He had a guest role in the television
**predicted**: @-@ , and , . , also a appearance seller in the series , and the , was the by a in in the film , . by . who was released by the . the Academy of in was previously appearance in the film series

When I used another frameworks for language modelling (e.g. TensorFlow, Keras) and the same function to predict the next words, it gives me more meaningful results.

Maybe I should modify your code in some way for this task?

Thanks.
